packages/service-library/src/servicelib/long_running_tasks/task.py

```diff
@@ -397,33 +448,30 @@ async def remove_task(
                 raise
             return
 
-        if tracked_task.task_id in self._created_tasks:
-            task_to_cancel = self._created_tasks.pop(tracked_task.task_id, None)
-            if task_to_cancel is not None:
-                # canceling the task affects the worker that started it.
-                # awaiting the cancelled task is a must since if the CancelledError
-                # was intercepted, those actions need to finish
-                await cancel_wait_task(task_to_cancel)
+        await self._tasks_data.mark_task_for_removal(
+            tracked_task.task_id, tracked_task.task_context
+        )
 
-            await self._tasks_data.delete_task_data(task_id)
+        if not wait_for_removal:
+            return
 
         # wait for task to be removed since it might not have been running
         # in this process
         async for attempt in AsyncRetrying(
-            wait=wait_exponential(max=2),
+            wait=wait_exponential(max=1),
             stop=stop_after_delay(_TASK_REMOVAL_MAX_WAIT),
             retry=retry_if_exception_type(TryAgain),
```

Comment:
```suggestion
            retry=retry_unless_exception_type(TaskNotFoundError),
```

Reply:
then you can remove the suppress

Reply:
So you still have to use a suppress for TaskNotFoundError since you wait for the error to occur. Tenacity does not suppress it. Regardless I use this proposed method since it's more readable